design/ceph/object/ceph-cosi-driver.md

```diff
@@ -0,0 +1,87 @@
+# Ceph COSI Driver Support
+
+## Targeted for v1.12
+
+## Background
+
+Container Object Storage Interface (COSI) is a specification for container orchestration frameworks to manage object storage. Even though there is no standard procotol defined for Object Store, it has flexibility to add support for all. The COSI spec abstracts common storage features such as create/delete buckets, grant/revoke access to buckets, attach/detach buckets, and more. COSI released v1alpha1 with Kubernetes 1.25. COSI is a new project and is not yet fully integrated by Kubernetes.
+More details about COSI can be found [here](https://kubernetes.io/blog/2022/09/02/cosi-kubernetes-object-storage-management/)
+It is projected that COSI will be the only supported object storage driver in the near future. In-tree drivers such as Ceph RGW will be replaced with their respective COSI drivers.
+
+## Current Status of Ceph COSI Driver
+
+The [Ceph COSI driver](https://github.com/ceph/ceph-cosi) is currently in the pre-alpha status. It is currently being tested with latest COSI Spec and APIs with images:
```

Comment:
The focus of the design doc is how we want the feature to work, I'm not sure we need to mention the current state of the pre-alpha driver.

Reply:
Sure will remove this para